snooty/parser.py

```diff
@@ -530,25 +530,27 @@ def dispatch_departure(self, node: tinydocutils.nodes.Node) -> None:
             popped.children = [n.Section((node.get_line(),), popped.children)]
 
     def handle_facet(self, node: rstparser.directive, line: int) -> None:
-        try:
-            ref: Union[rstparser.directive, tinydocutils.nodes.Element] = node
-            facet_str_pairs: List[tuple[str, str]] = [
-                (ref["options"]["name"], ref["options"]["values"])
-            ]
-
-            while ref.parent and ref.parent.get("name") == "facet":
-                ref = ref.parent
-                facet_str_pairs.append(
-                    (ref["options"]["name"], ref["options"]["values"])
-                )
+        ref: Union[rstparser.directive, tinydocutils.nodes.Element] = node
+        for value in ref["options"]["values"].split(","):
+            single_value = value.strip()
+            try:
+                facet_str_pairs: List[tuple[str, str]] = [
+                    (ref["options"]["name"], single_value)
+                ]
+
+                while ref.parent and ref.parent.get("name") == "facet":
```

Comment:
Hey Seung! I think there might be a bug here. If a child facet node has multiple values, on the second iteration, ref will no longer be defined and we won't enter the while loop. Because of this, the second value of the nested facet will only have a single tuple containing the nested facet category name and value, but not the parent. For example:

```rst
.. facet::
    :name: target_product
    :values: atlas

    .. facet::
        :name: sub_product
        :values: charts, atlas-cli
```

When checking the `sub_product` value `atlas-cli`, `ref` will be undefined because in the previous iteration we keep loop until we have `ref.parent`

Reply:
Just to confirm, I pulled your branch down and added a unit test for the facet that I specified in the last post. I did receive an error:

Users/brandonly/Documents/gitrepos/snooty-parser/snooty/test_parser.py::test_valid_facets failed: def test_valid_facets() -> None:
        path = FileId("test.rst")
        project_config = ProjectConfig(ROOT_PATH, "")
        parser = rstparser.Parser(project_config, JSONVisitor)
    
        page, diagnostics = parse_rst(
            parser,
            path,
            """
    
    .. facet::
        :name: target_product
        :values: atlas
    
        .. facet::
            :name: sub_product
            :values: charts, atlas-cli
    
    """,
        )
        page.finish(diagnostics)
>       assert len(diagnostics) == 0
E       AssertionError: assert 1 == 0
E        +  where 1 = len([MissingFacet('Facet specified is not a valid facet: sub_product:atlas-cli', (6, 0))])

snooty/test_parser.py:3678: AssertionError

Reply:
One more comment, I think this might be a solution:

```python
        for value in node["options"]["values"].split(","):
            ref: Union[rstparser.directive, tinydocutils.nodes.Element] = node
```

We can reset the `ref` value for every iteration so that we don't mutate between iterations of the for loop. Tested it locally and it looks like the tests pass.

Reply:
thanks for the catch @branberry ! i added the small bit you suggested as the solution

i added the [test case to postprocess](https://github.com/mongodb/snooty-parser/pull/524/commits/4691464d71dd8640305c5d65b12bb4cdcb415b47#diff-bfac8a18bf662c13854408d2116c9293dc488fde36361670d1c7af27becdafa3L2896) (and had to make some changes) since this will be more verbose of actually constructing the output. PTAL !